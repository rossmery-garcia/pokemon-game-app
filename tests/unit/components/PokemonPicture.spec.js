import { shallowMount } from '@vue/test-utils';
import PokemonPicture from '@/components/PokemonPicture';

describe('PokemonPicture component', () => {

  test('Should match with the snapshot', () => {

    const wrapper = shallowMount( PokemonPicture, {
      props: {
        pokemonId: 1,
        showPokemon: false
      }
    });

    expect( wrapper.html() ).toMatchSnapshot();
  });

  test('Should show the hidden image and the pokemon 100', () => {

    const wrapper = shallowMount( PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false
      }
    });

    const [ img1, img2 ] = wrapper.findAll('img');

    expect( img1.exists() ).toBeTruthy();
    expect( img2 ).toBe(undefined);
    expect( img1.classes('hidden-pokemon') ).toBe(true); //-- or toBeTruthy() --

    expect( img1.attributes('src') ).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')

  });

  test('Should show the pokemon picture if showPokemon:true', () => {

    const wrapper = shallowMount( PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: true
      }
    });

    const img = wrapper.find('img');

    expect( img.exists() ).toBeTruthy();

    expect( img.classes('hidden-pokemon') ).toBe(false);
    expect( img.classes('fade-in') ).toBe(true);

  });
})