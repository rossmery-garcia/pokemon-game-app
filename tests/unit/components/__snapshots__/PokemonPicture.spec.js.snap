// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture component Should match with the snapshot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="hidden-pokemon" alt="Pokemon image"></div>`;
